bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceDialog.java

```diff
@@ -898,35 +898,58 @@ private void storeSettings() {
 		writeConfiguration();
 	}
 
+	/**
+	 * Initializes the string to search for and the according text in the find field
+	 * based on either the selection in the target or, if nothing is selected, with
+	 * the newest search history entry.
+	 */
+	private void initFindString() {
+		if (!okToUse(fFindField)) {
+			return;
+		}
+
+		fFindField.removeModifyListener(fFindModifyListener);
+		if (!initFindStringFromSelection()) {
+			initFindStringFromHistory();
+		}
+		fFindField.setSelection(new Point(0, fFindField.getText().length()));
+		fFindField.addModifyListener(fFindModifyListener);
+	}
+
 	/**
 	 * Initializes the string to search for and the appropriate text in the Find
 	 * field based on the selection found in the action's target.
 	 */
-	private void initFindStringFromSelection() {
+	private boolean initFindStringFromSelection() {
 		String fullSelection = getCurrentSelection();
-		if (fullSelection != null && okToUse(fFindField)) {
-			boolean isRegEx = findReplaceLogic.isRegExSearchAvailableAndActive();
-			fFindField.removeModifyListener(fFindModifyListener);
-			if (!fullSelection.isEmpty()) {
-				String firstLine = getFirstLine(fullSelection);
-				String pattern = isRegEx ? FindReplaceDocumentAdapter.escapeForRegExPattern(fullSelection) : firstLine;
-				fFindField.setText(pattern);
-				if (!firstLine.equals(fullSelection)) {
-					// multiple lines selected
-					findReplaceLogic.deactivate(SearchOptions.GLOBAL);
-					fGlobalRadioButton.setSelection(false);
-					fSelectedRangeRadioButton.setSelection(true);
-				}
-			} else {
-				if ("".equals(fFindField.getText())) { //$NON-NLS-1$
-					if (!findHistory.isEmpty())
-						fFindField.setText(findHistory.get(0));
-					else
-						fFindField.setText(""); //$NON-NLS-1$
-				}
-			}
-			fFindField.setSelection(new Point(0, fFindField.getText().length()));
-			fFindField.addModifyListener(fFindModifyListener);
+		if (fullSelection == null || fullSelection.isEmpty()) {
+			return false;
+		}
+
+		String searchInput = getFirstLine(fullSelection);
+		boolean isSingleLineInput = searchInput.equals(fullSelection);
+		if (findReplaceLogic.isRegExSearchAvailableAndActive()) {
+			searchInput = FindReplaceDocumentAdapter.escapeForRegExPattern(fullSelection);
+		}
+		fFindField.setText(searchInput);
+
+		if (isSingleLineInput) {
+			// initialize search with current selection to allow for execution of replace
+			// operations
+			findReplaceLogic.findAndSelect(findReplaceLogic.getTarget().getSelection().x, fFindField.getText());
+		} else {
+			fGlobalRadioButton.setSelection(false);
+			fSelectedRangeRadioButton.setSelection(true);
+		}
+		return true;
+	}
+
+	private void initFindStringFromHistory() {
+		if ("".equals(fFindField.getText())) { //$NON-NLS-1$
+			if (!findHistory.isEmpty())
```

Comment:
In general, we wanted to always add braces even for single-line-scopes.
Wouldn't it make sense to introduce the braces for these blocks now, while we are already doing a refactoring?

Reply:
Done in 033648c.